### PR TITLE
Fix 2x Out-of-bounds and 2x NPE in searchGpxRoute

### PR DESCRIPTION
**Pull request by RZR-UA**

### CODE REVIEW

File: OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java [hunk 1]
```diff
@@ -1141,7 +1145,11 @@ private RouteCalcResult searchRouteAndPrepareTurns(final RoutingContext ctx, Lat
 			ctx.finalRouteSegment = new BinaryRoutePlanner().searchRouteInternal(local, s, e, null);
 			result = RouteResultPreparation.convertFinalSegmentToResults(ctx, ctx.finalRouteSegment);
 			addPrecalculatedToResult(recalculationEnd, result);
-			makeStartEndPointsPrecise(result, s.getPreciseLatLon(), e.getPreciseLatLon());
+			if (ctx.calculationProgress.totalApproximateDistance == 0) {
```
Comment (vshcherb) [inline OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java:1148]: Previous code is cleaner and doesn't rely on strange calculationProgress

Comment (RZR-UA) [reply OsmAnd-java/src/main/java/net/osmand/router/RoutePlannerFrontEnd.java]: 70b15bc592

**Comment by RZR-UA** [comment]
https://github.com/osmandapp/OsmAnd-Issues/issues/2412

**Comment by vshcherb** [comment]
https://github.com/osmandapp/OsmAnd/commit/b84319bcfa8aff6fd0e68fc29cdbdcfdaf413e15

**Pull request closed without merging**